Applications/Spire/Source/Ui/QuantityBox.cpp

```diff
@@ -14,14 +16,65 @@ namespace {
   };
 }
 
+struct QuantityBox::UnsignedQuantityModel : OptionalQuantityModel {
+  mutable CurrentSignal m_current_signal;
+  std::shared_ptr<OptionalQuantityModel> m_model;
+  scoped_connection m_current_connection;
+
+  UnsignedQuantityModel(std::shared_ptr<OptionalQuantityModel> model)
+    : m_model(std::move(model)),
+      m_current_connection(m_model->connect_current_signal(
+        [=] (const auto& current) {
+          on_current(current);
+        })) {}
+
+  optional<Quantity> get_minimum() const override {
+    return 0;
+  }
+
+  optional<Quantity> get_maximum() const override {
+    return m_model->get_maximum();
+  }
+
+  Quantity get_increment() const override {
+    return m_model->get_increment();
+  }
+
+  QValidator::State get_state() const override {
+    return m_model->get_state();
+  }
+
+  const optional<Quantity>& get_current() const override {
+    return m_model->get_current();
+  }
+
+  QValidator::State set_current(const optional<Quantity>& value) override {
+    if(value && *value < 0) {
+      return m_model->set_current(Quantity(0));
+    }
+    return m_model->set_current(value);
+  }
+
+  connection connect_current_signal(
+      const CurrentSignal::slot_type& slot) const override {
+    return m_current_signal.connect(slot);
+  }
+
+  void on_current(const optional<Quantity>& current) {
```

Comment:
Is there a reason this method and the other members (connection, model, etc.) aren't private?

Reply:
I think the class is only used by the QuantityBox and its definition is in the source file, so other classes can't access it. In this case, there is no difference between private and public.